cmd/zed/agents/zfs_agents.c

```diff
@@ -57,9 +57,11 @@ typedef struct guid_search {
 	uint64_t	gs_pool_guid;
 	uint64_t	gs_vdev_guid;
 	char		*gs_devid;
+	boolean_t	is_spare;
+	boolean_t	is_l2arc;
```

Comment:
Note: these could probably be a single variable keeping track of the device type.